File: widest-vertical-area-between-two-points-containing-no-points.ts
{
    const maxWidthOfVerticalArea = function(points) {
        let sorted: number[] = [];
        let wide = 0;

        for(let i = 0; i < points.length; i++) {
            sorted.push(points[i][0])
        }
        
        sorted.sort((a,b) => a - b)
        
        for(let i = 1; i < sorted.length; i++) {
            wide = Math.max(wide, Math.abs(sorted[i - 1] - sorted[i]))
            console.log(wide)
        }

        return wide
    };

    console.log(maxWidthOfVerticalArea([[8,7],[9,9],[7,4],[9,7]]))

}

File: widest-vertical-area-between-two-points-containing-no-points.rs
use std::cmp;

pub fn max_width_of_vertical_area(points: Vec<Vec<i32>>) -> i32 {
    let mut sorted = Vec::new();
    let mut wide = 0;

    for i in 0..points.len() {
        sorted.push(points[i][0]);
    }

    sorted.sort();

    for i in 1..sorted.len() {
        wide = cmp::max(wide, (sorted[i - 1] - sorted[i]).abs())
    }

    wide
}
